app/models/endpoint.rb

```diff
@@ -80,6 +80,7 @@ def self.seed_archive_endpoints_from_config(preservation_policies)
         endpoint.endpoint_node = endpoint_config.endpoint_node
         endpoint.storage_location = endpoint_config.storage_location
         endpoint.preservation_policies = preservation_policies
+        endpoint.delivery_class = Endpoint.delivery_classes[endpoint_config.delivery_class.constantize]
```

Comment:
i didn't expect the extra indirection of referring to the enum mapping, but when i just set equal to `endpoint_config.delivery_class.constantize`, i got this error in the specs:

```
ActiveRecord::StatementInvalid:
  PG::InvalidTextRepresentation: ERROR:  invalid input syntax for integer: "S3WestDeliveryJob"
  : INSERT INTO "endpoints" ("endpoint_name", "created_at", "updated_at", "endpoint_node", "storage_location", "endpoint_type_id", "delivery_class") VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING "id"
```

which makes me think AR is trying to cast the class to a string before doing the enum lookup and assignment?  i found that slightly unexpected.  anyway, i'd greatly prefer this indirection so that we can put the delivery class name in the settings, because putting raw int values in the settings seems error prone, and counter to the purpose of human readable enums.

would welcome suggestions that simplify this.

Reply:
I get the same behavior.  Your way is fine.  The idea of having actual classes as the enum keys is probably subverting some of the rails internals.  Like we definitely can't use the enum to do `endpoint.(::S3WestDeliveryJob)!` the same way we can do `pres_copy.ok!`.  

Maybe we should have strings in the enum and a wrapping getter that does the `constantize`?  I think that would make the guts work more conventionally.  But not a huge deal to me, since we will only ever be setting this value in one place.

Reply:
filed #890 as backlog tech debt